src/pages/ProcessDetail.tsx

```diff
@@ -79,47 +81,110 @@ class ProcessDetail extends React.PureComponent<IProps, IState> {
+            selectedTab: selectedTab,
+            product: productById(enrichedProcess.product, products),
+        });
+        this.context.apiClient
+            .processSubscriptionsByProcessId(enrichedProcess.id)
+            .then((res) => {
+                this.setState({ subscriptionProcesses: res, loaded: true });
+            })
+            .catch((err) => {
+                if (err.response && err.response.status === 404) {
+                    this.setState({ notFound: true, loaded: true });
+                } else {
+                    throw err;
+                }
             });
-            this.context.apiClient
-                .processSubscriptionsByProcessId(enrichedProcess.id)
-                .then((res) => {
-                    this.setState({ subscriptionProcesses: res, loaded: true });
-                })
-                .catch((err) => {
-                    if (err.response && err.response.status === 404) {
-                        this.setState({ notFound: true, loaded: true });
-                    } else {
-                        throw err;
+    };
+
+    updateProcessStep = (step: Step) => {
+        const enrichedProcess = this.state.process as CustomProcessWithDetails;
+        const stepUserInput: InputForm | undefined = step.form;
+        const tabs = stepUserInput ? this.state.tabs : ["process"];
+        const selectedTab = stepUserInput ? "user_input" : "process";
+
+        this.setState({
+            process: {
+                ...enrichedProcess,
+                steps: enrichedProcess.steps.map((process_step) => {
+                    if (process_step.name === step.name) {
+                        return { ...process_step, ...step };
                     }
-                });
+                    return process_step;
+                }),
+            },
+            stepUserInput: stepUserInput,
+            tabs: tabs,
+            selectedTab: selectedTab,
         });
     };
 
+    handleWebsocketError = (error: any) => {
+        setFlash(error.detail, "error");
+    };
+
+    componentDidMount = () => {
+        const client = websocketService.connect(`api/processes/test/${this.props.match.params.id}`);
+        this.setState({ client: client });
+        client.onmessage = ({ data }) => {
+            if (typeof data === "string") {
+                try {
+                    const json = JSON.parse(data);
+                    const process = json.process;
+                    const step: Step = json.step;
+                    const error = json.error;
+                    if (process) {
+                        this.initializeProcessDetails(process);
+                    }
+                    if (step) {
+                        this.updateProcessStep(step);
+                    }
+                    if (error) {
+                        this.handleWebsocketError(error);
+                    }
+                } catch (e) {
+                    console.error("invalid JSON");
```

Comment:
Haven't looked into detail, but don't we have a logger for that (that might eventually dump to the console, Sentry, both...)

Reply:
Sentry will automatically catch these exceptions. You could add a post to the user-error endpoint.